documentation/src/docs/asciidoc/user-guide/migration-from-junit4.adoc

```diff
@@ -55,21 +55,122 @@ concurrent test execution. It can be enabled and configured using the following
+ForkJoinPool-1-worker-2 - FooTest::test2
+ForkJoinPool-1-worker-1 - BarTest::test3
+ForkJoinPool-1-worker-2 - FooTest::test3
+----
+
+[[migrating-from-junit4-parallel-execution-method-level]]
+==== Parallelization at Method Level
+
+Alternatively, we can enable the parallel test execution at a `method` level,
+rather than the `class` level:
+
+[source,properties]
+----
+junit.vintage.execution.parallel.enabled=true
 junit.vintage.execution.parallel.methods=true
 junit.vintage.execution.parallel.pool-size=4
 ----
 
-With these properties set, the `VintageTestEngine` will execute tests in parallel,
-potentially significantly reducing the overall test suite execution time.
+Therefore, the test methods from within a class will be executed in parallel,
+while different test classes will be executed sequentially:
+
+[source,plaintext]
+----
+ForkJoinPool-1-worker-1 - BarTest::test1
+ForkJoinPool-1-worker-2 - BarTest::test2
+ForkJoinPool-1-worker-3 - BarTest::test3
+
+ForkJoinPool-1-worker-3 - FooTest::test1
+ForkJoinPool-1-worker-2 - FooTest::test2
+ForkJoinPool-1-worker-1 - FooTest::test3
+----
+
+[[migrating-from-junit4-parallel-execution-class-and-method-level]]
+==== Full Parallelization
+
+Finally, we can also enable the parallelization at both `class` and `method` level:
+
+[source,properties]
+----
+junit.vintage.execution.parallel.enabled=true
+junit.vintage.execution.parallel.classes=true
+junit.vintage.execution.parallel.methods=true
+junit.vintage.execution.parallel.pool-size=8
+----
+
+With these properties set, the `VintageTestEngine` will execute all the tests classes
+and methods in parallel, potentially significantly reducing the overall test suite execution time:
+
+[source,plaintext]
+----
+ForkJoinPool-1-worker-6 - FooTest::test2
+ForkJoinPool-1-worker-7 - BarTest::test3
+ForkJoinPool-1-worker-3 - FooTest::test1
+ForkJoinPool-1-worker-8 - FooTest::test3
+ForkJoinPool-1-worker-5 - BarTest::test2
+ForkJoinPool-1-worker-4 - BarTest::test1
+----
+
+[[migrating-from-junit4-parallel-execution-disabled]]
+==== Sequential Execution
+
+On the other hand, if we disable the parallel execution, the `VintageTestEngine`
+will execute the tests sequentially, regardless of the other properties:
+
+[source,properties]
+----
+junit.vintage.execution.parallel.enabled=false
+junit.vintage.execution.parallel.classes=true
+junit.vintage.execution.parallel.methods=true
+----
+
+Similarly, the parallelization won't work if we only enable the parallel execution,
+but we don't specifically enable a parallel execution level:
+
+[source,properties]
+----
+junit.vintage.execution.parallel.enabled=true
+junit.vintage.execution.parallel.classes=false
+junit.vintage.execution.parallel.methods=false
+----
+
+In such cases, all the tests will be executed sequentially:
+[source,plaintext]
+----
+main - BarTest::test1
+main - BarTest::test2
+main - BarTest::test3
+
+main - FooTest::test1
+main - FooTest::test2
+main - FooTest::test3
+----
```

Comment:
This is actually be considered a configuration error and produces a warning. Therefore, we should point that out and omit the example below.

```suggestion
Similarly, tests will be executed sequentially if you enable parallel execution in general
but enable neither class-level nor method-level parallelization.
```

Reply:
removed